apps/sensenet/src/components/view-controls/permission-view.tsx

```diff
@@ -0,0 +1,216 @@
+import { PathHelper } from '@sensenet/client-utils'
+import { GenericContent } from '@sensenet/default-content-types'
+import { useLogger, useRepository } from '@sensenet/hooks-react'
+import {
+  Avatar,
+  Button,
+  Collapse,
+  createStyles,
+  Link,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  makeStyles,
+  Theme,
+} from '@material-ui/core'
+import ExpandLess from '@material-ui/icons/ExpandLess'
+import ExpandMore from '@material-ui/icons/ExpandMore'
+import GroupOutlined from '@material-ui/icons/GroupOutlined'
+import clsx from 'clsx'
+import React, { useContext, useEffect, useState } from 'react'
+import { useHistory } from 'react-router'
+import { ResponsivePersonalSettings } from '../../context'
+import { useGlobalStyles } from '../../globalStyles'
+import { useLocalization } from '../../hooks'
+import { getUrlForContent } from '../../services'
+import { AclResponseType } from './permission-types'
+
+const useStyles = makeStyles((theme: Theme) => {
+  return createStyles({
+    permissionEditorContainer: {
+      margin: '0px 80px',
+    },
+    titleContainer: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      marginTop: '30px',
+    },
+    title: {
+      fontSize: '20px',
+    },
+    textBolder: {
+      fontWeight: 500,
+    },
+    listTitle: {
+      marginLeft: '16px',
+    },
+    iconWrapper: {
+      height: '40px',
+      width: '40px',
+    },
+    collapseWrapper: {
+      paddingLeft: '30px',
+    },
+    anchor: {
+      fontSize: '14px',
+      color: theme.palette.primary.main,
+      paddingLeft: '15px',
+      cursor: 'pointer',
+      '&:hover': {
+        textDecoration: 'underline',
+      },
+    },
+  })
+})
+
+export interface PermissionViewProps {
+  contentPath: string
+}
+
+export const PermissionView: React.FC<PermissionViewProps> = (props) => {
+  const classes = useStyles()
+  const globalClasses = useGlobalStyles()
+  const repo = useRepository()
+  const localization = useLocalization()
+  const history = useHistory()
+  const logger = useLogger('PermissionEditor')
+  const uiSettings = useContext(ResponsivePersonalSettings)
+  const [permissions, setPermissions] = useState<AclResponseType | undefined>(undefined)
+  const [currentContent, setCurrentContent] = useState<GenericContent | undefined>()
+  const [openInheritedList, setOpenInheritedList] = useState<boolean>(false)
+  const [openSetOnThisList, setOpenSetOnThisList] = useState<boolean>(true)
+
+  useEffect(() => {
+    async function getCurrentContent() {
+      const result = await repo.load({
+        idOrPath: props.contentPath,
+      })
+      setCurrentContent(result.d)
+    }
+    getCurrentContent()
+  }, [props.contentPath, repo])
+
+  useEffect(() => {
+    async function getAllPermissions() {
+      try {
+        const result = await repo.executeAction<any, AclResponseType>({
```

Comment:
I think GetAcl should be moved to `repository.security` in the client-core package